node_api/src/services/entity.service.ts

```diff
@@ -301,4 +301,37 @@ export default class EntityService {
       })
     ).data;
   }
+
+  async findEntitiesBySnomedCodes(codes: string[]) {
+    const iris = codes.map(code => SNOMED.NAMESPACE + code);
+    const result = await this.getPartialEntities(iris, [RDFS.LABEL, IM.CODE]);
+    return result.map(resolved => resolved.data);
+  }
+
+  async findValidatedEntitiesBySnomedCodes(codes: string[]) {
+    const entities = await this.findEntitiesBySnomedCodes(codes);
+    const needed = await this.getDistillation(entities as TTIriRef[]);
+    const response = [] as any[];
+    for (const entity of entities) {
+      const isInvalid = isObjectHasKeys(entity, ["@id"]) && !isObjectHasKeys(entity, [RDFS.LABEL, IM.CODE]);
+      const index = needed.findIndex(neededEntity => neededEntity["@id"] === entity["@id"]);
+      const isIncluded = response.some(added => entity["@id"] === added["@id"]);
+      if (isInvalid) {
+        entity.statusCode = "Invalid";
```

Comment:
validationCode is more fitting name, no confusion with existing status functionality